js/RoadMap.js

```diff
@@ -62,47 +63,53 @@ var RoadMap = (function() {
     });
 
     // Add all markers to the Map
-    var addMarkers = (function() {
-        console.log('RoadMap', 'addMarkers');
-        var stations = stationStorage.getStations();
+    var addMarkers = (function(_stationStorage) {
+        console.log('RoadMap', 'addMarkers()');
+        stationStorage = _stationStorage;
+
         var myIcon = '';
+        var markersList = [];
 
-        for(var i=0; i < stations.length; i++) {
+        // build Markers
+        for(var i=0; i < stationStorage.length; i++) {
```

Comment:
Oh, there is a `stationStorage = stationStorage.getStations()`.
Please avoid reusing the same variable name when data type changes.
In the right-hand member, `stationStorage` is a `StationStorage` (this is why it is called like so). But in the left-hand member, it is a list of stations.

It is really confusing actually, and `stationStorage` is not a good name for a list of stations, as it is already associated with another type of object. You could name it `stations` or `stationList` for instance.

Furthermore, it breaks a global variable (within the Black Box scope) which could be used in some other places as the original instance of `StationStorage`. This is an unexpected side effect.